x/ecocredit/server/core/bridge_receive_test.go

```diff
@@ -0,0 +1,316 @@
+			Jurisdiction: "US-KY",
+			Metadata:     "hi",
+			ClassId:      "C01",
+		},
+	}
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+	assert.Equal(t, res.ProjectId, project.Id)
+	assert.Equal(t, res.BatchDenom, batch.Denom)
+
+	// this was a fresh account, so we know their balance is only what was bridged to it.
+	bal, err := s.k.Balance(s.ctx, &core.QueryBalanceRequest{
+		Address:    recipient,
+		BatchDenom: batch.Denom,
+	})
+	assert.NilError(t, err)
+	balAfter := bal.Balance
+	assert.Equal(t, balAfter.TradableAmount, "3")
+}
+
+func TestBridgeReceive_ProjectNoBatch(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	recipient := testutil.GenAddress()
+	refId := "VCS-001"
+	project, batch := setupBridgeTest(s, refId)
+	startDate, endDate := (&timestamppb.Timestamp{Seconds: 500}).AsTime(), (&timestamppb.Timestamp{Seconds: 600}).AsTime()
+
+	msg := core.MsgBridgeReceive{
+		Issuer: s.addr.String(),
+		Batch: &core.MsgBridgeReceive_Batch{
+			Recipient: recipient,
+			Amount:    "3",
+			OriginTx: &core.OriginTx{
+				Id:     "0x12345",
+				Source: "polygon:0x12345",
+			},
+			StartDate: &startDate,
+			EndDate:   &endDate,
+			Metadata:  "hi",
+			Note:      "bridged test",
+		},
+		Project: &core.MsgBridgeReceive_Project{
+			ReferenceId:  refId,
+			Jurisdiction: "US-KY",
+			ClassId:      "C01",
+		},
+	}
+
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+	assert.Equal(t, res.ProjectId, project.Id)
+	assert.Check(t, res.BatchDenom != batch.Denom)
+
+	batch, err = s.stateStore.BatchTable().GetByDenom(s.ctx, res.BatchDenom)
+	assert.NilError(t, err)
+
+	bal, err := s.k.Balance(s.ctx, &core.QueryBalanceRequest{
+		Address:    recipient,
+		BatchDenom: batch.Denom,
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, bal.Balance.TradableAmount, msg.Batch.Amount)
+}
+
+func TestBridgeReceive_None(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	setupBridgeTest(s, "VCS-002")
+	recipient := testutil.GenAddress()
+	start, end := time.Now(), time.Now()
+	msg := core.MsgBridgeReceive{
+		Issuer: s.addr.String(),
+		Batch: &core.MsgBridgeReceive_Batch{
+			Recipient: recipient,
+			Amount:    "3",
+			OriginTx: &core.OriginTx{
+				Id:     "0x12345",
+				Source: "polygon:0x12345",
+			},
+			StartDate: &start,
+			EndDate:   &end,
+			Metadata:  "bar",
+			Note:      "bridged",
+		},
+		Project: &core.MsgBridgeReceive_Project{
+			ReferenceId:  "VCS-001",
+			Jurisdiction: "US-KY",
+			Metadata:     "foo",
+			ClassId:      "C01",
+		},
+	}
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+
+	batch, err := s.stateStore.BatchTable().GetByDenom(s.ctx, res.BatchDenom)
+	assert.NilError(t, err)
+	bal, err := s.k.Balance(s.ctx, &core.QueryBalanceRequest{
+		Address:    recipient,
+		BatchDenom: batch.Denom,
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, bal.Balance.TradableAmount, msg.Batch.Amount)
+}
+
+func TestBridgeReceive_TooManyBatches(t *testing.T) {
```

Comment:
Not sure I'm understanding this test scenario. Isn't it the same as `TestBridgeReceive_ChoosesLatestBatch()` below? Aren't we not erroring any more on multiple matching batches?

Reply:
ah yeah this test should be stale now, removed it in https://github.com/regen-network/regen-ledger/pull/1174/commits/47f604c0b5a1d3b7efde98adf016ce462672fc7b